realm-jni/src/util.cpp

```diff
@@ -118,8 +118,8 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             break;
 
         case Unspecified:
-            jExceptionClass = env->FindClass("java/lang/RuntimeException");
-            message = "Unspecified exception. " + classStr;
+            jExceptionClass = env->FindClass("io/realm/exceptions/RealmError");
+            message = "Unrecoverable error condition. " + classStr;
```

Comment:
We should not try to parse text in the exception  - that's a no go. Then we need more detailed exceptions from core instead.